pkg/batchfetcher/batchfetcher.go

```diff
@@ -53,14 +53,14 @@ func NewModule(mc *ModuleConfig, epochNr t.EpochNr, clientProgress *clientprogre
 	// The DeliverCert handler requests the transactions referenced by the received availability certificate
 	// from the availability layer.
 	// TODO: Make sure the certificate has been verified by the producer of this event.
-	dsl.UponEvent[*eventpb.Event_DeliverCert](m, func(cert *eventpb.DeliverCert) error {
+	eventpbdsl.UponDeliverCert(m, func(sn t.SeqNr, cert *apbtypes.Cert) error {
 		// Create an empty output item and enqueue it immediately.
 		// Actual output will be delayed until the transactions have been received.
 		// This is necessary to preserve the order of incoming and outgoing events.
 		item := outputItem{event: nil}
 		output.Enqueue(&item)
 
-		if cert.Cert.Type == nil {
+		if cert.Pb().Type == nil {
```

Comment:
This is something I have been thinking about it myself. I will add the check as it is a (small) optimization, but I think we should open a TODO/issue for a more standard way to check for empty certificates (in the view change it is a `[]byte{}`, in the availability module it is `[]byte{0}` and here it seems to be a nil. This is not immediate though as it might even make sense to have different empty certs depending on which module should check.